pkg/sypgp/sypgp.go

```diff
@@ -327,14 +327,22 @@ func RemovePubKey(toDelete string) error {
 
 	var newKeyList []openpgp.Entity
 
+	matchKey := false
+
 	// sort through them, and remove any that match toDelete
 	for i := range elist {
 		// if the elist[i] dose not match toDelete, then add it to newKeyList
-		if !compareLocalPubKey(elist[i], toDelete) {
+		if !CompareKeyEntity(elist[i], toDelete) {
 			newKeyList = append(newKeyList, *elist[i])
+		} else {
+			matchKey = true
```

Comment:
if `matchKey` is true, shouldn't it stop looping arount the list of entities `elist`? if that's so I would add a `break` after `matchKey = true`

Reply:
No, this is only used to test if theres was a match with the *fingerprint_to_delete*, and the local fingerprint.

Reply:
Yea but if it matches shouldn't it stop looking for it?

Reply:
In this case no, because we are building up `newKeyList` and dont want to lose it, this will result in missing public keys if the function should succeed.